### PR TITLE
Fix vint warnings and overall code style

### PR DESCRIPTION
**Pull request by hauleth**

### CODE REVIEW

File: autoload/asyncomplete.vim [hunk 1]
```diff
@@ -244,27 +238,15 @@ function! s:get_active_sources_for_buffer() abort
...
-if exists('*matchstrpos')
-    function! s:matchstrpos(expr, pattern) abort
-        return matchstrpos(a:expr, a:pattern)
-    endfunction
-else
-    function! s:matchstrpos(expr, pattern) abort
-        return [matchstr(a:expr, a:pattern), match(a:expr, a:pattern), matchend(a:expr, a:pattern)]
-    endfunction
-endif
-
```
Comment (hauleth) [inline autoload/asyncomplete.vim]: Not needed as `matchstrpos` was added in [Patch 7.4.1685](http://ftp.vim.org/vim/patches/7.4/7.4.1685) so each version that has async support has this function.

File: autoload/asyncomplete.vim [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+    echohl ErrorMsg
+    echomsg 'Vim/Neovim compiled with timers required for asyncomplete.vim.'
+    echohl NONE
+    if has('nvim')
+        call asyncomplete#log('neovim compiled with timers required.')
+    else
+        call asyncomplete#log('vim compiled with timers required.')
+    endif
+    finish
+endif
```
Comment (hauleth) [inline autoload/asyncomplete.vim]: Moved to the top as rest of the file is irrelevant without `timers`.

File: autoload/asyncomplete.vim [hunk 3]
```diff
@@ -112,16 +111,11 @@ function! asyncomplete#context() abort
     let l:ret['col'] = l:ret['curpos'][2]
     let l:ret['filetype'] = &filetype
     let l:ret['filepath'] = expand('%:p')
-    if l:ret['filepath'] == ''
-        " this is necessary here, otherwise empty filepath is somehow
-        " converted to None in vim's python binding.
-        let l:ret['filepath'] = ""
-    endif
```
Comment (hauleth) [inline autoload/asyncomplete.vim]: We do not use Python, so this fragment does nothing.

File: autoload/asyncomplete.vim (hunk 1 above)
Comment (prabirshrestha) [reply autoload/asyncomplete.vim]: Old neovim doesn’t have it so I wouldn’t remove it yet.

File: autoload/asyncomplete.vim [hunk 4]
```diff
@@ -55,7 +58,7 @@ function! asyncomplete#register_source(info) abort
         execute 'augroup asyncomplete_source_event_' . a:info['name']
         for l:event in a:info['events']
             let l:exec =  'if get(b:,"asyncomplete_enable",0) | call s:python_cm_event("' . a:info['name'] . '", "'.l:event.'",asyncomplete#context()) | endif'
-            if type(l:event) == type('')
+            if type(l:event) == v:t_string
```
Comment (prabirshrestha) [inline autoload/asyncomplete.vim]: older vim/neovim doesn't have support for this so I stopped using v:t_ and instead use `type` function only.

**Comment by hauleth** [comment]
@prabirshrestha I have fixed all issues and rebased code that it will now contain one functional commit.

**Comment by prabirshrestha** [comment]
@hauleth looks good expect that you seem to have removed default values in some places. Can we add it back? `get(b:, 'asyncomplete_enable')`.

Now that github has squash I can use it directly so no need to worry about squashing on your side.

**Comment by hauleth** [comment]
@prabirshrestha because default value is not needed, as default returned value is `0` if nothing else is specified.

```
get({list}, {idx} [, {default}])			*get()*
		Get item {idx} from |List| {list}.  When this item is not
		available return {default}.  Return zero when {default} is
		omitted.
get({dict}, {key} [, {default}])
		Get item with key {key} from |Dictionary| {dict}.  When this
		item is not available return {default}.  Return zero when
		{default} is omitted.
```

File: autoload/asyncomplete.vim [hunk 5]
```diff
@@ -1,3 +1,15 @@
+if !has('timers')
+    echohl ErrorMsg
+    echomsg 'Vim/Neovim compiled with timers required for asyncomplete.vim.'
+    echohl NONE
+    if has('nvim')
+        call asyncomplete#log('neovim compiled with timers required.')
```
Comment (hauleth) [inline autoload/asyncomplete.vim]: Also I think this line should be changed to:

> Newer version of NeoVim is required. This doesn't support timers.

As NeoVim doesn't have "compilation flags" to enable or disable features. So only way that someone doesn't have it when using version earlier than 0.1.5.

**Pull request merged**